apm-sniffer/apm-sdk-plugin/kafka-plugin/src/main/java/org/apache/skywalking/apm/plugin/kafka/define/KafkaProducerPropertiesInstrumentation.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.apm.plugin.kafka.define;
+
+import net.bytebuddy.description.method.MethodDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.ConstructorInterceptPoint;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.InstanceMethodsInterceptPoint;
+import org.apache.skywalking.apm.agent.core.plugin.match.ClassMatch;
+
+import static org.apache.skywalking.apm.agent.core.plugin.bytebuddy.ArgumentTypeNameMatch.takesArgumentWithType;
+import static org.apache.skywalking.apm.agent.core.plugin.match.NameMatch.byName;
+
+/**
+ * after version 2.1.0 use Properties to config
+ */
+public class KafkaProducerPropertiesInstrumentation extends AbstractKafkaInstrumentation {
+
+    public static final String ENHANCE_CLASS = "org.apache.kafka.clients.producer.KafkaProducer";
+    public static final String CONSTRUCTOR_INTERCEPTOR_CLASS = "org.apache.skywalking.apm.plugin.kafka.ProducerConstructorPropertiesInterceptor";
+    public static final String CONSTRUCTOR_INTERCEPTOR_FLAG = "java.util.Properties";
+
+    @Override
+    public ConstructorInterceptPoint[] getConstructorsInterceptPoints() {
+        return new ConstructorInterceptPoint[]{
+                new ConstructorInterceptPoint() {
+                    @Override
+                    public ElementMatcher<MethodDescription> getConstructorMatcher() {
+                        return takesArgumentWithType(0, CONSTRUCTOR_INTERCEPTOR_FLAG);
```

Comment:
I checked the `KafkaProducer` class in the version 2.3.1 (that you mentioned in the issue), and found that this constructor should be instrumented by `KafkaProducerMapInstrumentation`, this seems to be duplicated. Can you elaborate more? Thanks 

![image](https://user-images.githubusercontent.com/15965696/97663055-87846100-1ab3-11eb-82bc-bfefbead0fc6.png)

Reply:
KafkaProducerMapInstrumentation  
which version add this class

Reply:
i use 8.1.0 , can not support kafka2.3.1

Reply:
> KafkaProducerMapInstrumentation. which version add this class

It's there for a long time.

> i use 8.1.0 , can not support kafka2.3.1
 
So did it work after your changes in this PR?

Reply:
yes

Reply:
@hsoftxl do you use a `List` as `bootstrap.servers` in the properties?

Reply:
I've rechecked 8.1.0, it should intercept the constructor you mentioned in this PR, and I believe the instrumentation you added in this PR is duplicated.

If you're using a `List` as `bootstrap.servers` in the properties, it has problem in 8.1.0 but if you rebuild the plugin from the `master` branch (with or without your changes in this PR), it should work, can you confirm?

Reply:
![image](https://user-images.githubusercontent.com/6965803/97841617-2bbf0f80-1d21-11eb-9186-26a15e4e7d91.png)
my properies

i rebuild the plugin from master, when i rebuild the plugin whitout my change , it has exception  , list / string case exception,  i will test one more time.

Reply:
![image](https://user-images.githubusercontent.com/6965803/97841617-2bbf0f80-1d21-11eb-9186-26a15e4e7d91.png)
my properies

i rebuild the plugin from master, when i rebuild the plugin whitout my change , it has exception  , list / string case exception,  i will test one more time.

![image](https://user-images.githubusercontent.com/6965803/97842221-3cbc5080-1d22-11eb-9a84-2aca7093dc02.png)

Reply:
@kezhenxu94 mybey  the first time id rebulid the plugin is not base on master.
i will check again

Reply:
![image](https://user-images.githubusercontent.com/15965696/97842743-0e8b4080-1d23-11eb-9f6e-82c42bf65e05.png)

Make sure you have the latest codes, the master branch have this type detection, but from the exception log, seems it's not the case

Reply:
the master code is well,       i builded the plugin  from 8.1.0 tag